src/contexts/OracleDataContext.tsx

```diff
@@ -103,6 +104,22 @@ type ProcessAssertionsAction = DispatchAction<"assertions", Assertions>;
 
 type DispatchActions = ProcessRequestsAction | ProcessAssertionsAction;
 
+function mergeData(
```

Comment:
this is so we can properly merge more information, need to dedupe and union the values from previous + new state